### DISCUSSION TITLE
What is matdbg and how do I use it?

### DISCUSSION THREAD

**Discussion by redstonedash** [google/filament#4979] [Q&A]
In this issue matdbg is mentioned as a way to debug materials https://github.com/google/filament/issues/4976#issuecomment-995020693 . This is exactly what I need to do right now, but I can't find anything in the repo called matdbg.

**Comment by romainguy** [comment]
It's part of the debug builds of filament. It allows to connect to a running instance via the web browser to inspect and modify shaders at runtime. Note that it lets you modify the shaders generated by matc, not the source material files (although that's something I'd like us to add eventually).

**Reply by redstonedash** [reply]
Thank you! Are you able to connect it to filament for the web though?
Also I see in the README you posted that conversion to typescript is in the plans for the future. Type script has a nice feature (which I use for my project) where types can be signified with comments so the files can remain just normal JavaScript files. A recent change to filament actually made this easier, because filament is now a namespace in the .d.ts file you no longer need webpack or to actually use typescript at all in order to get all of the advantages of typescript.

**Comment by romainguy** [comment]
All the info about matdbg is here btw: https://github.com/google/filament/blob/main/libs/matdbg/README.md
